src/typelib.erl

```diff
@@ -25,6 +25,14 @@ pp_type(Type) ->
     %	File      -> S ++ " in " ++ File
     %end.
 
+pp_intersection_type([]) ->
+    "";
+%% TODO: pp_type seems to have problems printing bounded types.
```

Comment:
yes, bounded funs can only appear in specs but not in types (in the erlang AST) so they are not part of `erl_parse:abstract_type()`.  maybe we could extend `typchecker:type()` or have another type that includes it

Reply:
Btw, this function is just cut and paste from typechecker.erl so if we want to fix the problem of having it let's do that in a follow up PR.

Reply:
Indeed @gomoripeti, we need to distinguish between toplevel function specs and types which fall under `erl_parse:abstract_type()`. In particular, we need proper pretty printing of the former. The code here is just a hack that I made so that the typechecker wouldn't crash.